code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -338,6 +338,10 @@
 		addtimer(cb, 1)
 		addtimer(cb, 4)
 		addtimer(cb, 6)
+		if(rapid == 2)
```

Comment:
This could be re-written to shoot first and then check a `switch(rapid)`, then modify a `var/extra_shots` and run a `for` using it.